### ISSUE TITLE
Is it possible to override MP3 duration?

### ISSUE THREAD

**Issue by glureau** [google/ExoPlayer#9616]
I've a mp3 file that I can't modify, and it defines a bad duration (2:00 instead of 35+ minutes).

https://d3ctxlq1ktw2nl.cloudfront.net/staging/2021-9-25/d51792a0-e335-f459-375f-a092656d308b.mp3

Is it possible to ignore the duration or avoid the player to re-start playing from 0:00?
I'm seeing some players like Youtube that continue playing even if the UI is broken (still displaying 2:00 but sound is ok).
Note: I know the real duration of the file.

**Comment by ojw28** [comment]
I tried playing this file in a bunch of places (ExoPlayer, Chrome, VLC) and in all cases the duration is 37:18, which is presumably the correct duration. So it's unclear to me what the problem is. Please clarify.

**Comment by glureau** [comment]
Youtube Music (v3.45.54) displays 01:59, progress bar is at 100% after the 2 first minutes but song keeps playing after the 100% is reached.

Files by Google (v1.0.400489743) displays 02:00, same for progress bar but song is stopped at 2:00, like if it was the end of the file. 

The app Musique (pre-installed app, v45.6.4.0_f30b9fd_20210817) displays 01:59, but keeps playing (same behaviour than Youtube Music).

Just downloaded MX Player from MX Media (v1.40.9), also displaying 01:59 and keeps playing.

Chrome and VLC are fine.

I'm using ExoPlayer 2.14.0, the player stops after 2min and restart playing the same track. ExoPlayer triggers a onPositionDiscontinuity at 02:00, and I think it's restarting the track at the same time. I would expect it to ignore the wrong duration and keeps playing the song.

Is it something possibly fixed in the 2.15.1? I didn't find a match in tickets but I may try an upgrade.

**Comment by ojw28** [comment]
I cannot reproduce this with ExoPlayer 2.15.1, ExoPlayer 2.14.0, or File by Google. The duration shows as 37:18 in all cases. Is it possible that the link above is not the one that you meant to post, or that it's serving a different file to me than it is to you?

**Comment by glureau** [comment]
I think we're getting the same file. That said I've download it locally before opening it on the apps. If you want to try, I get a file with MD5: e185d06ffe7233bfb90d25c0fbb7c49d

Some additional info if it can help:
Phone: OPPO Reno4 Z 5G (model: CPH2065)
OS: ColorOS v11.1 (Android 11)

As it was reproduced on some other phones in my company I thought it'll be quite easy to reproduce, looks like I'll try to dig deeper. I've tested an upgrade to 2.15.1, and still reproduce the issue. I'll try to build another app from scratch to ensure it's not my code, tell me if there is some other tests/checks that would make sense for you.

Thanks for your time!

EDIT: After checkout ExoPlayer on master branch, I replaced the "Google Play MP3" link to mine, and the exoplayer works great. That's probably on my side then, will update this issue if I found more stuff.
